src/common/types.ts

```diff
@@ -221,6 +222,21 @@ export type ServerInfoType = 'MemberCount';
 export type RoleTypeKey = keyof typeof RoleType;
 export type Maybe<T> = T | undefined | null;
 
+export type MessageSendData =
+  | string
+  | discord.MessagePayload
+  | (discord.ReplyMessageOptions & { split?: false }); // Discord v13 change
+
+export type MessageEditData = string | discord.MessagePayload;
+
+export function isSlashCommand(plugin: unknown): boolean {
```

Comment:
You can use a type guard and this will simplify the type casting you're doing in all the consumers that are calling this function

`export function isSlashCommand(plugin: unknown): plugin is ISlashPlugin`

Reply:
True